Adding maven auth support (#29)
actions/setup-java: README.md

      - run: java -cp java HelloWorldApp
```

Publising to an Apache Maven Repository:
```yaml
jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v1
    - name: Set up JDK 1.8

>       uses: actions/setup-java@master

Reviewer: ```suggestion
      uses: actions/setup-java@v1
```

      with:
        java-version: 1.8
        server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
        username: ${{ github.actor }} # username for server authentication
        password: ${{ github.token }} # password or token for authentication
    - name: Build with Maven
      run: mvn -B package --file pom.xml
    - name: Publish to GitHub Packages Apache Maven
      run: mvn deploy
    - name: Set up Apache Maven Central

>       uses: actions/setup-java@master

Reviewer: ```suggestion
      uses: actions/setup-java@v1
```

      with: # running setup-java again overwrites the settings.xml
        java-version: 1.8
        server-id: maven
        username: maven_username
        password: ${{ secrets.MAVEN_CENTRAL_TOKEN }} # password from secrets store

>     - name: Publish to Apache Maven Central

Reviewer: Could you also add an example using Gradle

Author: I was going to do Gradle in a follow up PR. Lemme try to wrap this up and if I can I'll add an example as well.

Reviewer: Another PR with a Gradle example is fine. As long as we eventually have one.

Author: Ok, added a Gradle example in 203550b
I have a demo build of gradle here https://github.com/pied-piper-inc/fib-gradle/commit/2b7855275aaa4153ba7f9fe64b3e9ef3b8272817/checks?check_suite_id=346412071

      - run: java -cp java HelloWorldApp
```

Publishing using Apache Maven:
```yaml
jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v1
    - name: Set up JDK 1.8
      uses: actions/setup-java@v1
      with:
        java-version: 1.8
        server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
        username: ${{ github.actor }} # username for server authentication
        password: ${{ github.token }} # password or token for authentication
    - name: Build with Maven
      run: mvn -B package --file pom.xml
    - name: Publish to GitHub Packages Apache Maven
      run: mvn deploy
    - name: Set up Apache Maven Central
      uses: actions/setup-java@v1
      with: # running setup-java again overwrites the settings.xml
        java-version: 1.8
        server-id: maven
        username: maven_username
        password: ${{ secrets.MAVEN_CENTRAL_TOKEN }} # password from secrets store
    - name: Publish to Apache Maven Central
      run: mvn deploy 
```

> ***NOTE: The `settings.xml` is created in the Actons $HOME directory by default and will overwrite existing files.***	

Reviewer: ```suggestion
***NOTE: The `settings.xml` file is created in the Actions $HOME directory. If you have an existing `settings.xml` file at that location, it will be overwritten***	
```
Small spelling fix. Also some rewording. Sounds a little scary right now as it seems like all existing files will be overwritten.

See the help docs on [Publishing a Package](https://help.github.com/en/github/managing-packages-with-github-packages/configuring-apache-maven-for-use-with-github-packages#publishing-a-package) for more information on the `pom.xml` file.

> Publishing using Gradle:

Reviewer: Good example 👍 Thanks for adding this!

        username: ${{ github.actor }} # username for server authentication
        password: ${{ github.token }} # password or token for authentication
    - name: Build with Maven
      run: mvn -B package --file pom.xml
    - name: Publish to GitHub Packages Apache Maven
      run: mvn deploy
    - name: Set up Apache Maven Central
      uses: actions/setup-java@v1
      with: # running setup-java again overwrites the settings.xml
        java-version: 1.8
        server-id: maven
        username: maven_username
        password: ${{ secrets.MAVEN_CENTRAL_TOKEN }} # password from secrets store
    - name: Publish to Apache Maven Central
      run: mvn deploy 
```

***NOTE: The `settings.xml` file is created in the Actions $HOME directory. If you have an existing `settings.xml` file at that location, it will be overwritten***	

See the help docs on [Publishing a Package](https://help.github.com/en/github/managing-packages-with-github-packages/configuring-apache-maven-for-use-with-github-packages#publishing-a-package) for more information on the `pom.xml` file.

## Publishing using Gradle
```yaml
jobs:

  build:
    runs-on: ubuntu-latest

    steps:

>     - uses: actions/checkout@v1

Author: I'll do some general spacing for clarity  and uniformity but I don't think a space between every step is required.

    - name: Set up JDK 1.8
      uses: actions/setup-java@v1

    - name: Build with Gradle
      run: gradle build

    - name: Publish to GitHub Packages
      run: gradle publish
      env:
        USERNAME: ${{ github.actor }}
        PASSWORD: ${{ secrets.GITHUB_TOKEN }}
```

***NOTE: The `USERNAME` and `PASSWORD` need to correspond to the credentials environment variables used in the publishing section of your `build.gradle`..***	

See the help docs on [Publishing a Package with Gradle](https://help.github.com/en/github/managing-packages-with-github-packages/configuring-gradle-for-use-with-github-packages#example-using-gradle-groovy-for-a-single-package-in-a-repository) for more information on the `build.gradle` configuration file.

> ## Apache Maven within a Shared Runner

Reviewer 2: Shared runner is not really the right term here.  It would be a self-hosted runner with multiple runners configured under the same user on the same machine.

When using an Actions shared runner the default `$HOME` directory can be shared by a number of workflows at the same time which could overwrite existing settings file. Setting the `m2-home` variable allows you to choose a unique location for your settings file.

```yaml
jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v1
    - name: Set up JDK 1.8 for Shared Runner
      uses: actions/setup-java@v1
      with:
        java-version: 1.8
        server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
        username: ${{ github.actor }} # username for server authentication
        password: ${{ github.token }} # password or token for authentication

>         m2-home: ${{ $GITHUB_WORKSPACE }} # location for the settings.xml file

Reviewer 2: I don't see `m2-home` in the `action.yml` below.

Author: Yeah, I switch that to the settings name and add it to the yaml file. 

AFAIK It is not possible to use the `$GITHUB_WORKSPACE` env to set this path. The only thing that works is using relative or absolute paths like `./config` or `/config/` respectively.

Reviewer 2: In your example above it would need to be `${{ github.workspace }}` . You can't reference a bash environment variable in the macro.
